src/plugins/mer/meremulatordevice.h

```diff
@@ -115,6 +114,10 @@ class MerEmulatorDevice : public MerDevice
 
     MerConnection *connection() const;
 
+    void addPortForwarding(const QString &ruleName, quint16 hostPort, quint16 emulatorVmPort) const;
+    bool removePortForwarding(const QString &ruleName);
+    bool hasPortForwarding(quint16 hostPort) const;
```

Comment:
Why did I care to explicitly state how exactly I would prefer it to look like? :) - Please fix according to https://github.com/sailfishos/sailfish-qtcreator/pull/165#discussion_r248649356

Reply:
Unfortunately, it is impossible to get information about the configured port forwarding rules by their names in the current architecture because the `VirtualMachineInfo` class contains only information about port numbers, not about rules names.

Do you propose to store the port forwarding rules names inside `VirtualMachineInfo` and use this information in the `hasPortForwarding()` method implementation?

Reply:
Yes I would do so. The user of the API should be able to verify that a port is forwarded according to their needs - by checking the ruleName.

Reply:
The `ruleName` in the API I suggested was an (optional) out-type parameter, so that you can determine the state with single call:

```
QString ruleName;
if (emulator->hasPortForwarding(1234, &ruleName) {
    if (ruleName == "rule1") {
        ... // OK, already set up
    } else {
        ... // Err, conflict
    }
} else {
    ... // Free, set it up
}
```